clients/metadata/device_service.go

```diff
@@ -41,58 +39,60 @@ type DeviceServiceClient interface {
 }
 
 type deviceServiceRestClient struct {
-	url      string
-	endpoint interfaces.Endpointer
+	client interfaces.RestClientBuilder
 }
 
 // NewDeviceServiceClient creates an instance of DeviceServiceClient
 func NewDeviceServiceClient(params types.EndpointParams, m interfaces.Endpointer) DeviceServiceClient {
-	s := deviceServiceRestClient{endpoint: m}
-	s.init(params)
+	s := deviceServiceRestClient{client: rest.ClientFactory(params, m)}
 	return &s
 }
 
-func (d *deviceServiceRestClient) init(params types.EndpointParams) {
-	if params.UseRegistry {
-		go func(ch chan string) {
-			for {
-				select {
-				case url := <-ch:
-					d.url = url
-				}
-			}
-		}(d.endpoint.Monitor(params))
-	} else {
-		d.url = params.Url
+func (dsc *deviceServiceRestClient) UpdateLastConnected(id string, time int64, ctx context.Context) error {
+	serviceURL, err := dsc.client.URLPrefix()
+	if err != nil {
+		return err
 	}
+
+	_, err = clients.PutRequest(serviceURL+"/"+id+"/lastconnected/"+strconv.FormatInt(time, 10), nil, ctx)
+	return err
 }
 
-// Helper method to request and decode a device service
-func (s *deviceServiceRestClient) requestDeviceService(url string, ctx context.Context) (models.DeviceService, error) {
```

Comment:
I removed this helper method because it is only used once and is unlikely to be used anywhere else.